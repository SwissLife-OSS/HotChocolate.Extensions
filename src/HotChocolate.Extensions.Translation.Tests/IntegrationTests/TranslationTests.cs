using System.Threading.Tasks;
using Microsoft.Extensions.DependencyInjection;
using HotChocolate;
using HotChocolate.Execution;
using HotChocolate.Types;
using Snapshooter.Xunit;
using Xunit;
using HotChocolate.Execution.Configuration;
using HotChocolate.Extensions.Translation.Tests.Mock;
using HotChocolate.Extensions.Translation.Resources;

namespace HotChocolate.Extensions.Translation.Tests.IntegrationTests
{
    public class TranslationTests
    {
        [Fact]
        public async Task Translate_WithNonNullableEnum_ShouldTranslateProperly()
        {
            //Arrange

            System.Threading.Thread.CurrentThread.CurrentCulture
                = new System.Globalization.CultureInfo("en");

            DummyValues keys = DummyValues.Bar;

            var query = "{ foo { key label } }";

            IRequestExecutorBuilder builder = new ServiceCollection()
                .AddSingleton<IResourcesProviderAdapter>(new EvergreenResourcesProviderAdapter())
                .AddGraphQL()
                .AddDirectiveType<TranslateDirectiveType<DummyValues>>()
                .AddQueryType(d =>
                    d.Field("foo")
                        .Resolver(keys)
                        .Translate<DummyValues>("prefix")
                );

            //Act
            IExecutionResult result = await builder.ExecuteRequestAsync(
                QueryRequestBuilder.New().SetQuery(query).Create());

            //Assert
            Snapshot.Match(result);
        }

        public enum DummyValues
        {
            Foo,
            Bar,
            Baz,
            Qux
        }
    }
}
