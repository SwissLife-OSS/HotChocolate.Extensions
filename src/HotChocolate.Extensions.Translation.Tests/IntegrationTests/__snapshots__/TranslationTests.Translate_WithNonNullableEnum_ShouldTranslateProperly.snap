{
  "Kind": "SingleResult",
  "RequestIndex": null,
  "VariableIndex": null,
  "Label": null,
  "Path": null,
  "Data": {
    "foo": {
      "key": "BAR",
      "label": "rms:prefix/Bar_English"
    }
  },
  "Items": null,
  "Errors": null,
  "Extensions": null,
  "Incremental": null,
  "ContextData": null,
  "HasNext": null,
  "IsDataSet": true
}
